lib/actor/Actor.d.ts

```diff
@@ -73,64 +81,65 @@ export declare type ValidActorMessageName = keyof ActorMessageType;
  * Users of this actor generally should not use {@link Actor#initPromise}. This
  * is an internal implementation detail for {@link hookup}.
  */
-export declare abstract class Actor<T> {
-    /**
-     * Do not use, it is an internal implementation detail used in {@link hookup}.
-     */
-    private readonly initPromise;
-    /**
-     * The name given to this actor by calling {@link hookup}.
-     */
-    actorName?: ValidActorMessageName;
-    constructor();
-    /**
-     * Init callback that can be used to perform some initialization logic.
-     * This method is invoked in the constructor of an {@link Actor} and should
-     * not be called by any user of the actor subclass.
-     *
-     * Note that no messages will be delivered until the resulting promise
-     * is resolved.
-     *
-     * @return A promise which resolves once this actor is initialized.
-     */
-    init(): Promise<void>;
-    /**
-     * Callback to process a message that was sent to this actor.
-     *
-     * Note that this callback is synchronous. This means that if an actor needs
-     * to perform expensive work (for example, encode an image), you need to
-     * perform this work asynchronously. You can make `onMessage` asynchronous
-     * if you want to use `await`. Note that message delivery will *not* be
-     * halted until `onMessage` as completed.
-     *
-     *    class MyActor extends Actor<MessageType> {
-     *      onMessage(message: MessageType) {
-     *        Promise.resolve().then(() => this.performExpensiveWork());
-     *      }
-     *
-     *      async performExpensiveWork() {
-     *        // Some long running task here
-     *      }
-     *    }
-     *
-     *
-     * For TypeScript users, this requires the specification
-     * of the {@link ActorMessageType}:
-     *
-     *    interface State {
-     *      count: number;
-     *    }
-     *
-     *    declare global {
-     *      interface ActorMessageType {
-     *        ui: State;
-     *      }
-     *    }
-     *
-     * @param message The message that was sent to this actor.
-     */
-    abstract onMessage(message: T): void;
-}
+export declare function actorMixin<T, S extends Constructable = Constructable<Object>>(superClass: S): {
+    new (...args: any[]): {
+        /**
+         * Do not use, it is an internal implementation detail used in {@link hookup}.
+         */
+        readonly initPromise: Promise<void>;
+        /**
+         * The name given to this actor by calling {@link hookup}.
+         */
+        actorName?: "ignoring" | "ignoring1" | "late" | undefined;
```

Comment:
This seems wrong, but I fear it is a TS issue rather than an "us" issue.

Reply:
Yeah, didn't write this myself. Assuming `tsc` is the culprit.

Reply:
Linked to upstream issue https://github.com/Microsoft/TypeScript/issues/27171